### DISCUSSION TITLE
LibreSBC as Loadbalancer, problem to setup

### DISCUSSION THREAD

**Discussion by poldy42** [hnimminh/libresbc#162] [Q&A] [answered]
Hi all,
I'm currently "playing around" with LibreSBC to set up a simple load balancer for a system that receives status messages in XML format via SIP messages.

I have defined two gateways, one outbound interconnections and one inbound interconnection, to which a fairly simple routing rule is attached (SIP profile, cluster and media class are also present).
For some reason the system is not routing the message to a gateway, but the message is being rejected by Freeswitch.
I am a little bit lost right now, I can't find a way to understand and to fix the error - every hint is highly aprreciated :-)

Best regards - Michael

----
tport.c:2769 tport_wakeup_pri() tport_wakeup_pri(0x14f16c004ec0): events IN
tport.c:2900 tport_recv_event() tport_recv_event(0x14f16c004ec0)
tport.c:3237 tport_recv_iovec() tport_recv_iovec(0x14f16c004ec0) msg 0x14f16c044f90 from (udp/10.0.30.10:5060) has 411 bytes, veclen = 1
recv 411 bytes from udp/[yy.yy.56.71]:52975 at 16:00:33.617243:

MESSAGE sip:xxx.xxx.193.210 SIP/2.0
Via: SIP/2.0/UDP 127.0.0.2:52975;branch=z9hG4bK.28eeab13;rport;alias
From: <sip:ip1@testsystem.eu>
To: <sip:ip1@xxx.xxx.193.210>
Call-ID: <TESTGPSSIP>
CSeq: 2 MESSAGE
Accept: text/plain, application/im-iscomposing+xml
Content-Type: application/scaip+xml
User-Agent: CLIP

<mrq><ref>21393</ref><cid>42420003</cid><dty>0003</dty><stc>0010</stc><crd>sip:</crd></mrq>

tport.c:3055 tport_deliver() tport_deliver(0x14f16c004ec0): msg 0x14f16c044f90 (411 bytes) from udp/yy.yy.56.71:5060/sip next=(nil)
nta.c:2957 agent_recv_request() nta: received MESSAGE sip:xxx.xxx.193.210 SIP/2.0 (CSeq 2) (load: 0 rps)
nta.c:3254 agent_check_request_via() nta: Via check: received=yy.yy.56.71
nta.c:3164 agent_recv_request() nta: MESSAGE (2) going to a default leg
nua_server.c:102 nua_stack_process_request() nua: nua_stack_process_request: entering
nua_stack.c:938 nh_create() nua: nh_create: entering
nua_common.c:108 nh_create_handle() nua: nh_create_handle: entering
nua_params.c:484 nua_stack_set_params() nua: nua_stack_set_params: entering
soa.c:278 soa_clone() soa_clone(static::0x14f16c001bd0, 0x14f16c0013d0, 0x14f16c04c6a0) called
soa.c:402 soa_set_params() soa_set_params(static::0x14f16c0500f0, ...) called
nta.c:4497 nta_leg_tcreate() nta_leg_tcreate(0x14f16c044850)
nua_stack.c:301 nua_stack_event() nua(0x14f16c04c6a0): event i_message 100 Trying
nua_stack.c:389 nua_application_event() nua: nua_application_event: entering
nua.c:1170 nua_handle_unref_user() nua nua_handle_unref_user(0x14f16c04c6a0): entering
nua_stack.c:558 nua_signal() nua(0x14f16c04c6a0): sent signal r_handle_unref
nua.c:1157 nua_unref_user() nua: nua_unref_user: entering
nua_stack.c:558 nua_signal() nua((nil)): sent signal r_unref
nua_stack.c:599 nua_stack_signal() nua(0x14f16c04c6a0): recv signal r_handle_unref
nua_stack.c:599 nua_stack_signal() nua((nil)): recv signal r_unref
nua.c:365 nua_handle_magic() nua: nua_handle_magic: entering
nua.c:878 nua_respond() nua: nua_respond: entering
nua_stack.c:558 nua_signal() nua(0x14f16c04c6a0): sent signal r_respond
nua.c:365 nua_handle_magic() nua: nua_handle_magic: entering
nua.c:920 nua_handle_destroy() nua: nua_handle_destroy: entering
nua_stack.c:558 nua_signal() nua(0x14f16c04c6a0): sent signal r_destroy
nua.c:1170 nua_handle_unref_user() nua nua_handle_unref_user(0x14f16c04c6a0): entering
nua_stack.c:558 nua_signal() nua(0x14f16c04c6a0): sent signal r_handle_unref
nua_stack.c:601 nua_stack_signal() nua(0x14f16c04c6a0): recv signal r_respond 202 Accepted
nua.c:1157 nua_unref_user() nua: nua_unref_user: entering
nua_stack.c:558 nua_signal() nua((nil)): sent signal r_unref
nua_params.c:484 nua_stack_set_params() nua: nua_stack_set_params: entering
soa.c:402 soa_set_params() soa_set_params(static::0x14f16c0500f0, ...) called
tport.c:3290 tport_tsend() tport_tsend(0x14f16c004ec0) tpn = UDP/yy.yy.56.71:52975
tport.c:4076 tport_resolve() tport_resolve addrinfo = yy.yy.56.71:52975
tport.c:4713 tport_by_addrinfo() tport_by_addrinfo(0x14f16c004ec0): not found by name UDP/yy.yy.56.71:52975
tport.c:3625 tport_vsend() tport_vsend(0x14f16c004ec0): 431 bytes of 431 to udp/yy.yy.56.71:52975
tport.c:3526 tport_send_msg() tport_vsend returned 431
send 431 bytes to udp/[yy.yy.56.71]:52975 at 16:00:33.617566:

SIP/2.0 202 Accepted
Via: SIP/2.0/UDP 127.0.0.2:52975;branch=z9hG4bK.28eeab13;rport=52975;alias;received=yy.yy.56.71
From: <sip:ip1@testsystem.eu>
To: <sip:ip1@xxx.xxx.193.210>;tag=K4NHj67HK2vBN
Call-ID: <TESTGPSSIP>
CSeq: 2 MESSAGE
Contact: <sip:10.0.30.10>
User-Agent: LibreSBC
Allow: INVITE, ACK, BYE, CANCEL, OPTIONS, MESSAGE, INFO, UPDATE, PRACK, NOTIFY
Supported: 100rel, timer, path, replaces
Content-Length: 0

nta.c:6879 incoming_reply() nta: sent 202 Accepted for MESSAGE (2)
nta.c:1371 set_timeout() nta: timer set to 32000 ms
nua_stack.c:599 nua_stack_signal() nua(0x14f16c04c6a0): recv signal r_destroy
nta.c:4550 nta_leg_destroy() nta_leg_destroy(0x14f16c044850)
soa.c:355 soa_destroy() soa_destroy(static::0x14f16c0500f0) called
nua_stack.c:599 nua_stack_signal() nua(0x14f16c04c6a0): recv signal r_handle_unref
nua_stack.c:599 nua_stack_signal() nua((nil)): recv signal r_unref
2024-06-10 16:00:33.603373 [DEBUG] sofia_presence.c:225 Can't find registered user ip1@xxx.xxx.193.210

----
GATEWAYS
=======

[       
    {
        "name": "DM90_DE_02",
        "desc": "2. DM90 DE",
        "ip": "10.0.140.163",
        "port": 5060,
        "transport": "udp"
    },
    {
        "name": "DM90_DE_01",
        "desc": "1. DM90 DE",
        "ip": "10.0.140.162",
        "port": 5060,
        "transport": "udp"
    }
]

OUTBOUND
========
[
    {
        "name": "DM90DE",
        "desc": "Connection DM90 DE",
        "sipprofile": "SP_DM90"
    }
]

{
    "nodes": [
        "_ALL_"
    ],
    "rtpaddrs": [
         "10.0.140.162/32",
        "10.0.140.163/32",
     ],
    "sipprofile": "SP_DM90",
    "cid_type": "none",
    "capacity_class": "small",
    "enable": true,
    "manipulation_classes": [],
    "sipaddrs": [],
    "distribution": "weight_based",
    "media_class": "MP_SCAIP",
    "nofailover_sip_codes": [],
    "privacy": [
        "none"
    ],
    "name": "DM90DE",
    "desc": "Connection DM90 DE",
    "translation_classes": [],
    "gateways": [
        {
            "name": "DM90_DE_01",
            "weight": 10
        },
        {
            "name": "DM90_DE_02",
            "weight": 5
        }
    ],
    "engagements": [
        "table:RouteAll"
    ]
}

INBOUND
======
[
    {
        "name": "IN_TESTSIP",
        "desc": "Connetion from Outside",
        "sipprofile": "SP_TESTSIP",
        "routing": "RouteAll"
    }
]

ROUTING
======
[
    {
        "name": "RouteAll",
        "desc": "route call to DM90DE",
        "action": "route",
        "routes": [
            "DM90DE",
            "DM90DE",
            "100"
        ]
    }
]

**Comment by hnimminh** [comment, marked as answer]
Hi @poldy42,
LibreSBC is for call only,  **MESSAGE** method does not support.  Kamailio might works for your case.
